Sources/SwiftCoAP/SCMessage.swift

```diff
@@ -203,7 +203,9 @@ public final class SCCoAPUDPTransportLayer {
         if let delegate = self.transportLayerDelegates[id] {
             delegate.delegate.transportLayerObject(self, didReceiveData: rawData, fromEndpoint: connection.endpoint)
             if delegate.observation == false && message.type == .acknowledgement {
-                self.transportLayerDelegates.removeValue(forKey: id)
+                operationsQueue.sync{ [weak self] in
```

Comment:
Can this be called from another `sync` block? If so it will cause deadlock

Reply:
Haven't seen the method is called from the same queue `sync` block somewhere but now I want to check the call stacks.